include/revng/StackAnalysis/ABI/CallingConventionTrait.h

```diff
@@ -0,0 +1,1089 @@
+  /// 8-16 bytes on 64-bit systems) are only allowed to start from a register
+  /// with an even index. For example, the ABI for a function like
+  /// ```
+  /// void function(uint32_t, uint64_t);
+  /// ```
+  /// on a system with 32-bit wide general purpose registers would lead to
+  /// the first argument being placed in the first argument register (say `r0`),
+  /// but then, because the second register is double-GRP-sized, its location
+  /// depends on this parameter. If `OnlyStartDoubleArgumentsFromAnEvenRegister`
+  /// is true, the argument is passed using a pair of registers starting from
+  /// the next one with an even index (`r2` and `r3` in this example),
+  /// otherwise, the next pair of registers is used (`r1` and `r2` here).
+  ///
+  /// \note: the option has no effect if `ArgumentsArePositionBased` is true.
+  static constexpr bool OnlyStartDoubleArgumentsFromAnEvenRegister = false;
+
+  /// States whether specified ABI allows splitting a single big (with size
+  /// exceeding the size of a single register) between the register and the
+  /// stack if there's not enough free registers to fit it
+  /// For example, let's say there's a big object of type `A` such that
+  /// `sizeof(A) == 16` bytes (on 32-bit architecture, that would mean that four
+  /// general purpose registers (`16 == 4 * 4`) are required to fit it).
+  /// And let's assume that the ABI in question has four GPRs available for
+  /// argument passing (`r0-r3`). Then, the ABI for a function like
+  /// ```
+  /// void function(uint32_t, A);
+  /// ```
+  /// would allocate the first argument into the `r0` register, but then
+  /// there would not be enough space to pass the second argument.
+  ///
+  /// In that case, if `ArgumentsCanBeSplitBetweenRegistersAndStack` is true,
+  /// the first 12 bytes of the second argument would still be placed into
+  /// `r1-r3` and the last 4 bytes would go on the stack.
+  /// Otherwise, the entire object would go on the stack with the remaining
+  /// three registers unused.
+  ///
+  /// \note: the option has no effect if `ArgumentsArePositionBased` is true.
+  static constexpr bool ArgumentsCanBeSplitBetweenRegistersAndStack = false;
+
+  /// Specifies how the stack arguments are passed. If it's true - they use
+  /// pointers-to-copy, otherwise, they are placed on the stack.
+  ///
+  /// \note: this only affects the arguments with size exceeding the size of
+  /// a single stack "slot" (which is equal to the stack alignment).
+  static constexpr bool UsePointerToCopyForStackArguments = false;
+
+  /// Specifies whether vector registers are always used in rigid pairs, meaning
+  /// that an odd-index vector register cannot be used without the even-index
+  /// register from the same pair being used to hold half the value.
+  ///
+  /// \note: The option still is not fully implemented since the vector register
+  /// support is still super limited.
+  static constexpr bool VectorRegistersArePaired = false;
```

Comment:
Unused, but so is the option about who cleans the stack.

It's just here so I don't need to look through all the docs again at the point where it'll become needed (and it will when we'll be adding vector register support).

Reply:
Uhm, maybe add a note about the fact that's there for future use?
Otherwise this is fine.

Reply:
What's the status for this?

Reply:
I temporarily removed this stuff and placed it into the [pad](https://pad.rev.ng/s/xtG2SPLiF). So this thread can be closed, I guess.